### PR TITLE
Import arcade props and targets

### PR DESCRIPTION
**Pull request by weshaggard**
Fixes https://github.com/dotnet/corefx/issues/32983

cc @ericstj @safern @ViktorHofer 

This is still dependent on a couple of PRs in arcade but I'm doing this to validate all the other OS legs and once that is done we can arrange the ordering of PR submission.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -39,7 +32,13 @@
   <!-- Provides package dependency version properties and verification/auto-upgrade configuration -->
   <Import Project="$(MSBuildThisFileDirectory)dependencies.props" />
 
+  <PropertyGroup>
+    <ProjectAssetsFile Condition="'$(ProjectAssetsFile)' == ''">dummy</ProjectAssetsFile>
```
Comment (ViktorHofer) [inline Directory.Build.props]: Why is that required?

File: external/test-runtime/XUnit.Runtime.depproj [hunk 2]
```diff
@@ -22,12 +22,12 @@
     <!-- Coverlet related properties -->
     <CoverletPackageId>coverlet.msbuild</CoverletPackageId>
     <CoverletPackageVersion>2.3.0</CoverletPackageVersion>
-    <!-- In case the project language is not set -->
-    <Language Condition="'$(Language)' == ''">unknown</Language>
+    <!-- Exlicitly set Language to c# to handle the anazlyzer resolution in nuget -->
```
Comment (ViktorHofer) [inline external/test-runtime/XUnit.Runtime.depproj]:
```suggestion
    <!-- Exlicitly set Language to c# to handle the analyzer resolution in nuget -->
```

Comment (wtgodbe) [reply external/test-runtime/XUnit.Runtime.depproj]: Also `Exlicitly` -> `Explicitly`

File: eng/versioning.targets [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+#endif
+]]>
+      </_NativeVersionFileContents>
+    </PropertyGroup>
+
+    <!--
+      Copy the NativeVersion.rc file next to the version header so that it can be picked
+      up and used in the native build along with the version.h file.
+    -->
+    <Copy SourceFiles="$(MSBuildThisFileDirectory)..\Tools\NativeVersion.rc"
```
Comment (safern) [inline eng/versioning.targets]: This seems to still be coming from buildtools, are we planning on moving it somewhere else? Should we generate this file on the fly just like version.c/h?

**Comment by weshaggard** [comment]
test Linux arm64 Release Build

File: Directory.Build.props (hunk 1 above)
Comment (ericstj) [reply Directory.Build.props]: Can you name this `unused` instead?

File: Directory.Build.props [hunk 4]
```diff
@@ -223,9 +225,36 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <_IsTestProjectBefore>$(IsTestProject)</_IsTestProjectBefore>
+    <IsPerformanceTestProject Condition="'$(IsPerformanceTestProject)' == '' AND '$(IsTestProject)' == 'true' AND $(MSBuildProjectName.EndsWith('Performance.Tests'))">true</IsPerformanceTestProject>
+  </PropertyGroup>
+
+  <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+
+    <!-- Need to reset our definition of IsTestProject after arcade sdk import -->
```
Comment (ericstj) [inline Directory.Build.props]: Is this still required?  I thought we were going to agree upon a convention, or allow to turn off arcade's convention.

File: Directory.Build.props [hunk 5]
```diff
@@ -286,6 +315,7 @@
     <WarningLevel>4</WarningLevel>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <Deterministic>false</Deterministic>
+    <CheckForOverflowUnderflow>false</CheckForOverflowUnderflow>
```
Comment (ericstj) [inline Directory.Build.props]: Can you add a comment here?  How bad was it when this was set to false?  Could we instead move this just to the individual projects that need it?

File: eng/Tools.props [hunk 6]
```diff
@@ -37,21 +42,28 @@
...
         Condition="!Exists('$(BuildToolsSemaphore)')"
         Command="$(InitToolsCmdLine) &gt; $(RepoRoot)init-tools.log" />
 
     <Touch Files="$(BuildToolsSemaphore)" AlwaysCreate="true" />
+    <OnError ExecuteTargets="DumpInitToolsLog" />
+  </Target>
+
+  <Target Name="DumpInitToolsLog">
+    <Message Text="Init-tools failed dumping log file $(RepoRoot)init-tools.log" />
+    <Exec Condition="Exists('$(RepoRoot)init-tools.log')" Command="type $(RepoRoot)init-tools.log" />
```
Comment (ericstj) [inline eng/Tools.props]: Does this work on Unix?  Maybe `<Message Text="$([System.IO.File]::ReadAllText('$(RepoRoot)init-tools.log'))" />`

File: eng/versioning.targets [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  <PropertyGroup Condition="'$(GenerateAssemblyInfo)'=='true'">
+    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)_AssemblyInfo.cs</AssemblyInfoFile>
+    <AssemblyInfoFile Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(IntermediateOutputPath)_AssemblyInfo.vb</AssemblyInfoFile>
+  </PropertyGroup>
+
+
+  <Target Name="DecideIfWeNeedToIncludeDllSafeSearchPathAttribute"
+    Condition="'$(IsDotNetFrameworkProductAssembly)' == 'true' AND '$(IsTestProject)' != 'true'">
+
+    <!-- We want to apply the IncludeDllSafeSearchPathAttribute on all DotNet assemblies (non test) that have a reference to System.Runtime.InteropServices -->
```
Comment (ericstj) [inline eng/versioning.targets]: Is this new behavior?  What about assemblies referencing netstandard or CoreLib?  Those *could* have it as well.  You could probably simplify this by scheduling it after ResolveReferences and before CoreCompile and just examining ReferencePath items instead.

File: src/Native/Windows/clrcompression/CMakeLists.txt [hunk 8]
```diff
@@ -81,7 +81,7 @@ if ($ENV{__VSVersion} STREQUAL vs2015)
 endif ()
 
 # Include 'bin/obj' dir since it contains _version.h
```
Comment (ericstj) [inline src/Native/Windows/clrcompression/CMakeLists.txt]: Update comment.

File: Directory.Build.props (hunk 4 above)
Comment (weshaggard) [reply Directory.Build.props]: Yes this is still required. @ViktorHofer will work on cleaning this up with his testing changes.

File: Directory.Build.props (hunk 5 above)
Comment (weshaggard) [reply Directory.Build.props]: It was pretty bad there were a large set of tests in lots of projects failing because this was turned on by default in arcade sdk. We have made attempts to enable it on some projects in the past but we decided against it.

File: eng/Tools.props (hunk 6 above)
Comment (weshaggard) [reply eng/Tools.props]: Good catch I'll update this.

File: eng/versioning.targets (hunk 7 above)
Comment (weshaggard) [reply eng/versioning.targets]: No this is existing behavior I just copied from BuildTools https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L43. We could probably clean this up but for now it is doing the job.

Comment (ericstj) [reply eng/versioning.targets]: I see.  I did a quick check of the shared framework and didn't notice anything that was missing this attribute but also using DllImport, so I guess we're ok.

File: Directory.Build.props (hunk 5 above)
Comment (ericstj) [reply Directory.Build.props]: Could we turn it on for just src projects?

File: external/test-runtime/XUnit.Runtime.depproj [hunk 9]
```diff
@@ -22,12 +22,12 @@
     <!-- Coverlet related properties -->
     <CoverletPackageId>coverlet.msbuild</CoverletPackageId>
     <CoverletPackageVersion>2.3.0</CoverletPackageVersion>
-    <!-- In case the project language is not set -->
-    <Language Condition="'$(Language)' == ''">unknown</Language>
+    <!-- Explicitly set Language to c# to handle the analyzer resolution in nuget -->
+    <Language Condition="'$(Language)' == ''">c#</Language>
```
Comment (safern) [inline external/test-runtime/XUnit.Runtime.depproj]: Is this still true? We're now going to inject the analyzers directly from the generated props file in Tools.props

File: Directory.Build.targets [hunk 10]
```diff
@@ -9,6 +9,24 @@
     <ImportDirectoryBuildTargets>false</ImportDirectoryBuildTargets>
   </PropertyGroup>
 
+  <!-- temporarily remap AssemblyKey to StrongNameKeyId -->
+  <PropertyGroup>
+    <StrongNameKeyId Condition="'$(AssemblyKey)' == 'MSFT'">Microsoft</StrongNameKeyId>
+    <StrongNameKeyId Condition="'$(AssemblyKey)' == 'ECMA'">ECMA</StrongNameKeyId>
+    <StrongNameKeyId Condition="'$(AssemblyKey)' == 'Open' or '$(IsTestProject)' == 'true'">Open</StrongNameKeyId>
+    <StrongNameKeyId Condition="'$(AssemblyKey)' == 'SilverlightPlatform'">SilverlightPlatform</StrongNameKeyId>
```
Comment (ViktorHofer) [inline Directory.Build.targets]: Is that for Silverlight assemblies? Are we still building some in corefx?

File: eng/Versions.props [hunk 11]
```diff
@@ -1,10 +1,20 @@
...
+    <MinorVersion>7</MinorVersion>
+
+    <!-- Always use shipping version insted of dummy version -->
+    <DotNetUseShippingVersions>true</DotNetUseShippingVersions>
+    <PreReleaseVersionLabel>preview1</PreReleaseVersionLabel>
+    <!-- Opt-out repo features -->
+    <UsingToolXliff>false</UsingToolXliff>
+
+    <!-- Always use shipping version insted of dummy version -->
+    <DotNetUseShippingVersions>true</DotNetUseShippingVersions>
```
Comment (ViktorHofer) [inline eng/Versions.props]: looks like copy paste overlook.

File: eng/versioning.targets [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    <PropertyGroup Condition="'$(IncludeDllSafeSearchPathAttribute)'==''">
+      <IncludeDllSafeSearchPathAttribute>false</IncludeDllSafeSearchPathAttribute> <!-- We don't need to include it by default. -->
+      <IncludeDllSafeSearchPathAttribute Condition="'%(ProjectReference.Filename)'=='System.Runtime.InteropServices'">true</IncludeDllSafeSearchPathAttribute>
+      <IncludeDllSafeSearchPathAttribute Condition="'%(Reference.Identity)'=='System.Runtime.InteropServices'">true</IncludeDllSafeSearchPathAttribute>
+    </PropertyGroup>
+  </Target>
+
+  <PropertyGroup>
+    <!-- corefx has never genreated these attributes so don't let the SDK generate them -->
```
Comment (ViktorHofer) [inline eng/versioning.targets]:
```suggestion
    <!-- corefx has never generated these attributes so don't let the SDK generate them -->
```

File: external/test-runtime/XUnit.Runtime.depproj (hunk 9 above)
Comment (ViktorHofer) [reply external/test-runtime/XUnit.Runtime.depproj]: Still we PackageReference xunit.analyzers here which needs the Language to be set.

Comment (safern) [reply external/test-runtime/XUnit.Runtime.depproj]: We can remove this reference and included into Tools.props so that the analyzers actually get injected to the compiler. Due to the infrastructure I added to the repo it has to be there. Also, they're not doing anything in this project, because this project is restored, but when building the tests that already happened, I don't see the xunit analyzers being passed to the compiler.

However this can be fixed in a follow up PR when we enable the analyzers for test projects.

Comment (ViktorHofer) [reply external/test-runtime/XUnit.Runtime.depproj]: > Also, they're not doing anything in this project

Right. We restore xunit.analyzers only as a dependency of xunit. We don't use it today.

> However this can be fixed in a follow up PR when we enable the analyzers for test projects.

I agree.

File: Directory.Build.targets (hunk 10 above)
Comment (weshaggard) [reply Directory.Build.targets]: I'm not sure if we are building any in corefx it might only be coreclr but I added the mapping for completeness. I should file an issue to remove this mapping and simply replace all the places in file because I didn't want to do that just yet.

Comment (weshaggard) [reply Directory.Build.targets]: Filed https://github.com/dotnet/corefx/issues/33112 for cleaning this up.

File: Directory.Build.props (hunk 5 above)
Comment (weshaggard) [reply Directory.Build.props]: Unfortunately not there are a lot of issues throughout and libraries that want to enable it should do it for just there projects.

File: eng/Versions.props (hunk 11 above)
Comment (weshaggard) [reply eng/Versions.props]: Not sure I follow this is an intentional change.

Comment (ViktorHofer) [reply eng/Versions.props]: see 7 lines above, you duplicated the property.

Comment (weshaggard) [reply eng/Versions.props]: Ah yes a merge issue I'll clean that up.

**Comment by weshaggard** [comment]
test Linux arm64 Release Build

**Comment by ViktorHofer** [comment]
Awesome!
